src/jaxsim/api/model.py

```diff
@@ -1931,11 +1932,17 @@ def step(
         ),
     )
 
+    exceptions.raise_if(
+        condition=t0_ns + jnp.array(dt * 1e9).astype(t0_ns.dtype) < t0_ns,
+        exception=OverflowError,
+        msg="The simulation time overflowed the maximum integer value. Consider using x64 by setting `JAX_ENABLE_X64=1`.",
+    )
```

Comment:
This is good enough for the time being. Though, I am wondering. Fixed step integrators do not need to have a well-formed $t_f$, and I mean that if internally the time overflows, the integrator still works as long as `dt` is finite. What about detecting the overflow after stepping, and reset `t0_ns` to 0?

Something like:

```py
tf_ns = t0_ns + jnp.array(dt * 1e9, dtype=t0_ns.dtype)
tf_ns = jnp.where(tf_ns >= t0_ns, tf_ns, jnp.array(0, dtype=t0_ns))
```

In addition, we might also add a `jax.lax.cond` with a `jax.debug.print` statement only in case of overflow, so that the user is notified in case they use the simulated time in their code.

Reply:
Solved after discussing f2f. We never raise the OverFlowError, we only print the debug statement.

Addressed in https://github.com/ami-iit/jaxsim/pull/225/commits/59370746f8ece3d931229122366fb60ae329faf5

Reply:
If we want, in the future we can perform the detection of overflow _before_ stepping the simulator, and in case of variable-step, raise the `OverFlowError`.